src/main/java/lotto/domain/Rank.java

```diff
@@ -21,18 +21,14 @@
         this.matchStatus = matchStatus;
     }
 
-    public static Rank getMatchResultWithBonusNumber(int total) {
-        if (total == MATCH_FIVE_AND_BONUS_NUMBERS.matchCount) {
+    public static Rank getMatchResult(int total, boolean isMatchWithBonusNumber) {
+        if (isMatchWithBonusNumber && total == MATCH_FIVE_AND_BONUS_NUMBERS.matchCount) {
             return MATCH_FIVE_AND_BONUS_NUMBERS;
         }
-        return getMatchResult(total);
-    }
-
-    public static Rank getMatchResult(int total) {
         return Arrays.stream(Rank.values())
                 .filter(rank -> rank.matchCount == total)
                 .findFirst()
-                .orElseGet(() -> Rank.valueOf(MATCH_ZERO_NUMBERS.name()));
+                .orElseGet(() -> Rank.valueOf(MATCH_MISS.name()));
```

Comment:
```suggestion
                .orElse(MATCH_MISS);
```

특정값을 돌려주기 때문에 위 코드도 괜찮을 것 같아요 :)